File: kmeans_openmp/Point.cpp
#include "Point.h"

Point::Point(int id, int x, int y) {
	id_point = id;
	this->x = x;
	this->y = y;

	id_cluster = -1;
}

Point::Point(int x, int y) {
	this->id_point = -1;
	this->x = x;
	this->y = y;

	id_cluster = -1;
}

void Point::setCluster(int id_cluster) {
	this->id_cluster = id_cluster;
}

int Point::getID() { return id_point;  }

int Point::getCluster() { return id_cluster; }

int Point::getX() { return x; }

int Point::getY() { return y; }

void Point::setX(int x) { this->x = x; }

void Point::setY(int y) { this->y = y; }

double Point::getDistance(Point other) {
        return sqrt(pow(this->x - other.getX(), 2)+ pow(this->y - other.getY(), 2));
}

File: kmeans_java/Point.java
import java.util.List;

public class Point {
	private int idPoint;
	private int x;
	private int y;
	private int idCluster;
  
	public Point(int id, int x, int y) {
		this.idPoint = id;
		this.x = x;
		this.y = y;
      
		idCluster = -1;
	}
  
	public Point(int x, int y) {
		this.idPoint = -1;
		this.x = x;
		this.y = y;
		
		idCluster = -1;
	}
	
	public double getDistance(Point other) {
        return Math.sqrt(Math.pow(this.x - other.x, 2)+ Math.pow(this.y - other.y, 2));
    }
	
	public static Point getMean(List<Point> points) {
        int sumX = 0;
        int sumY = 0;
        if (points.size() == 0) return new Point(sumX, sumY);
        for (Point point : points) {
            sumX += point.x;
            sumY += point.y;
        }
        return new Point(sumX / points.size(), sumY / points.size());
    }
  
	public void setCluster(int id) { this.idCluster = id; }
  
	public int getCluster() { return this.idCluster; }
  
	public int getID() { return this.idPoint; }
  
	public void setX(int x) { this.x = x; }
  
	public void setY(int y) { this.y = y; }
  
	public int getX() { return this.x; }
  
	public int getY() { return this.y; }
  
	@Override
	public String toString() {
		return "[x:" + this.x + ", y:" + this.y + ", pointId: "+this.idPoint 
				+ ", clusterId: "+ this.idCluster+ "]";
    }
      
	@Override
    public boolean equals(Object obj) {
        Point other = (Point) obj;
        return this.x == other.x && this.y == other.y;
	}
      
}